crypto/http/http_client.c

```diff
@@ -0,0 +1,1238 @@
+        if (n <= 0) {
+            if (BIO_should_retry(rctx->mem))
+                goto next_io;
+            rctx->state = OHS_ERROR;
+            return 0;
+        }
+
+        /* Don't allow excessive lines */
+        if (n == rctx->iobuflen) {
+            HTTPerr(0, HTTP_R_RESPONSE_LINE_TOO_LONG);
+            rctx->state = OHS_ERROR;
+            return 0;
+        }
+
+        /* First line */
+        if (rctx->state == OHS_FIRSTLINE) {
+            switch (parse_http_line1((char *)rctx->iobuf)) {
+            case HTTP_STATUS_CODE_OK:
+                rctx->state = OHS_HEADERS;
+                goto next_line;
+            case HTTP_STATUS_CODE_MOVED_PERMANENTLY:
+            case HTTP_STATUS_CODE_FOUND: /* i.e., moved temporarily */
+                if (rctx->method_GET) {
+                    rctx->state = OHS_REDIRECT;
+                    goto next_line;
+                }
+                HTTPerr(0, HTTP_R_REDIRECTION_NOT_ENABLED);
+                /* redirection is not supported/recommended for POST */
+                /* fall through */
+            default:
+                rctx->state = OHS_ERROR;
+                return 0;
+            }
+        }
+        key = (char *)rctx->iobuf;
+        value = strchr(key, ':');
+        if (value != NULL) {
+            *(value++) = '\0';
+            while (ossl_isspace(*value))
+                value++;
+            line_end = strchr(value, '\r');
+            if (line_end == NULL)
+                line_end = strchr(value, '\n');
+            if (line_end != NULL)
+                *line_end = '\0';
+        }
+        if (value != NULL && line_end != NULL) {
+            if (rctx->state == OHS_REDIRECT && strcmp(key, "Location") == 0) {
+                rctx->redirection_url = value;
+                return 0;
+            }
+            if (rctx->expected_ct != NULL && strcmp(key, "Content-Type") == 0) {
+                if (strcmp(rctx->expected_ct, value) != 0) {
+                    HTTPerr(0, HTTP_R_UNEXPECTED_CONTENT_TYPE);
+                    ERR_add_error_data(4, "expected=", rctx->expected_ct,
+                                       ",actual=", value);
+                    return 0;
+                }
+                rctx->expected_ct = NULL; /* content-type has been found */
+            }
+            if (strcmp(key, "Content-Length") == 0) {
+                resp_len = strtoul(value, &line_end, 10);
+                if (line_end == value || *line_end != '\0') {
+                    HTTPerr(0, HTTP_R_ERROR_PARSING_CONTENT_LENGTH);
+                    ERR_add_error_data(2, "input=", value);
+                    return 0;
+                }
+                if (!check_set_resp_len(rctx, resp_len))
+                    return 0;
+            }
+        }
+
+        /* Look for blank line: end of headers */
+        for (p = rctx->iobuf; *p; p++) {
+            if ((*p != '\r') && (*p != '\n'))
+                break;
+        }
+        if (*p) /* not end of headers */
```

Comment:
`*p != '\0'`

Reply:
This comparison coding style nit, as well as the further ones you mentioned plus a couple of similar further ones I've just found myself, had been inherited from `crypto/ocsp/ocsp_ht.c`.
I've fixed all of them.